File: src/main/java/botmanager/frostbalance/commands/influence/DailyRewardCommand.java
package botmanager.frostbalance.commands.influence;

import botmanager.Utilities;
import botmanager.frostbalance.Frostbalance;
import botmanager.frostbalance.Influence;
import botmanager.frostbalance.command.AuthorityLevel;
import botmanager.frostbalance.command.ContextLevel;
import botmanager.frostbalance.command.FrostbalanceGuildCommand;
import botmanager.frostbalance.command.GuildCommandContext;
import net.dv8tion.jda.api.entities.TextChannel;

import java.text.SimpleDateFormat;
import java.util.Date;

public class DailyRewardCommand extends FrostbalanceGuildCommand {

    SimpleDateFormat hours = new SimpleDateFormat("HH");
    
    public DailyRewardCommand(Frostbalance bot) {
        super(bot, new String[] {
                "daily"
        }, AuthorityLevel.GENERIC, ContextLevel.PUBLIC_MESSAGE);
    }

    @Override
    public void executeWithGuild(GuildCommandContext context, String[] params) {

        Influence gain = context.getMember().gainDailyInfluence();

        if (gain.getValue() > 0) {
            Utilities.sendGuildMessage((TextChannel) context.getChannel(), context.getMember().getEffectiveName() + ", your influence increases in " + context.getGuild().getName() + ".");

            Influence influence = context.getMember().getInfluence();
            context.sendPrivateResponse("You now have **" + String.format("%s", influence) + "** influence in **" + context.getGuild().getName() + "**.");
        } else {
            int hrsDelay = (24 - Integer.parseInt(hours.format(new Date())));
            context.sendResponse(context.getMember().getEffectiveName() + ", try again at midnight EST "
                    + "(around " + hrsDelay + " hour" + (hrsDelay > 1 ? "s" : "") + ").");
        }
    }

    @Override
    protected String info(AuthorityLevel authorityLevel, boolean isPublic) {
        if (isPublic) {
            return "**" + getBot().getPrefix() + "daily** - gives you all the influence you can get today, instantly";
        } else {
            return null;
        }
    }
}

File: src/main/java/botmanager/frostbalance/commands/influence/DailyRewardCommand.kt
package botmanager.frostbalance.commands.influence

import botmanager.frostbalance.Frostbalance
import botmanager.frostbalance.command.AuthorityLevel
import botmanager.frostbalance.command.ContextLevel
import botmanager.frostbalance.command.FrostbalanceGuildCommand
import botmanager.frostbalance.command.GuildMessageContext
import java.text.SimpleDateFormat
import java.util.*

class DailyRewardCommand(bot: Frostbalance?) : FrostbalanceGuildCommand(bot, arrayOf(
        "daily"
), AuthorityLevel.GENERIC, ContextLevel.PUBLIC_MESSAGE) {
    var hours = SimpleDateFormat("HH")
    public override fun executeWithGuild(context: GuildMessageContext, params: Array<String>) {

        if (!context.guild.allows(context.player)) {
            return context.sendResponse(context.guild.notAllowed)
        }

        val gain = context.member.gainDailyInfluence()
        if (gain.value > 0) {
            context.sendResponse(context.member.effectiveName + ", your influence increases in " + context.guild.name + ".")
            val influence = context.member.influence
            context.sendPrivateResponse("You now have **" + String.format("%s", influence) + "** influence in **" + context.guild.name + "**.")
        } else {
            val hrsDelay = 24 - hours.format(Date()).toInt()
            context.sendResponse(context.member.effectiveName + ", try again at midnight EST "
                    + "(around " + hrsDelay + " hour" + (if (hrsDelay > 1) "s" else "") + ").")
        }
    }

    override fun info(authorityLevel: AuthorityLevel?, isPublic: Boolean): String? {
        return if (isPublic) {
            "**" + bot.prefix + "daily** - gives you all the influence you can get today, instantly"
        } else {
            null
        }
    }
}